### PR TITLE
Use the new Placement dictionary in `Dropdown` and `Tooltip` #DS-923

### PR DESCRIPTION
**Pull request by adamkudrna**
image

https://jira.lmc.cz/browse/DS-923

### CODE REVIEW

File: packages/web-react/src/components/Dropdown/Dropdown.tsx [hunk 1]
```diff
@@ -2,14 +2,14 @@ import React, { createElement, useRef, LegacyRef } from 'react';
...
-import { DropdownPlacements, SpiritDropdownProps } from '../../types';
+import { SpiritDropdownProps } from '../../types';
 import { useDropdown } from './useDropdown';
 import { useDropdownStyleProps } from './useDropdownStyleProps';
 import { useDropdownAriaProps } from './useDropdownAriaProps';
 
 const defaultProps = {
-  placement: DropdownPlacements.BOTTOM_LEFT,
   enableAutoClose: true,
+  placement: 'bottom-left',
```
Comment (literat) [inline packages/web-react/src/components/Dropdown/Dropdown.tsx:12]:
```suggestion
  placement: Placements.BOTTOM_LEFT,
```
Please, the constant should be used here.

File: packages/web-react/src/components/Dropdown/useDropdownStyleProps.ts [hunk 2]
```diff
@@ -19,24 +20,15 @@ export interface UseDropdownStylePropsReturn {
 export const useDropdownStyleProps = (
   props: UseDropdownStyleProps = { isOpen: false },
 ): UseDropdownStylePropsReturn => {
-  const { isOpen, placement = DropdownPlacements.BOTTOM_LEFT, ...modifiedProps } = props;
+  const { isOpen, placement = 'bottom-left', ...modifiedProps } = props;
```
Comment (literat) [inline packages/web-react/src/components/Dropdown/useDropdownStyleProps.ts:23]:
```suggestion
  const { isOpen, placement = Placements.BOTTOM_LEFT, ...modifiedProps } = props;
```

File: packages/web-react/src/components/Tooltip/demo/TooltipPlacements.tsx [hunk 3]
```diff
@@ -1,46 +1,145 @@
...
-      </ButtonLink>
-      <Tooltip id="my-tooltip-hover-left" placement="left">
-        Hello there!
-      </Tooltip>
-    </TooltipWrapper>
-  </>
-);
+  return (
+    <form autoComplete="off" className="mx-auto">
+      <div className="Grid Grid--cols-3" style={{ alignItems: 'center', justifyItems: 'center' }}>
```
Comment (literat) [inline packages/web-react/src/components/Tooltip/demo/TooltipPlacements.tsx:17]: I think we should use a component here 🤔

File: packages/web-react/src/utils/__tests__/string.test.ts
Comment (literat) [inline packages/web-react/src/utils/__tests__/string.test.ts:1]: ❤️ Thanks.

File: packages/web-react/src/utils/string.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+/**
+ * Converts a kebab-case string to camelCase
+ * @param str
+ */
+export const kebabCaseToCamelCase = (str: string): string => {
+  return str.replace(/-([a-z])/g, (g) => g[1].toUpperCase());
```
Comment (literat) [inline packages/web-react/src/utils/string.ts:6]: Implicit return, please.

**Review by literat — COMMENTED** [review]
Good Job 👏 There are still several places where the dictionary can be used in code instead of hardcoded value. Please fix it and it will be ready to merge. Thanks a lot :-)

File: packages/web-react/src/components/Dropdown/__tests__/useDropdownStyleProps.test.ts [hunk 5]
```diff
@@ -1,35 +1,42 @@
 import { renderHook } from '@testing-library/react-hooks';
-import { DropdownPlacements } from '../../../types';
+import { PlacementDictionaryType } from '../../../types/shared/dictionaries';
```
Comment (crishpeen) [inline packages/web-react/src/components/Dropdown/__tests__/useDropdownStyleProps.test.ts:2]: does the path have to be this specific? Couldn't we import it from the types only?

File: packages/web-react/src/components/Dropdown/stories/Dropdown.stories.tsx [hunk 6]
```diff
@@ -2,7 +2,9 @@ import React, { Ref } from 'react';
 import { Markdown } from '@storybook/blocks';
 import type { Meta, StoryObj } from '@storybook/react';
 
-import { DropdownFullWidthModes, DropdownPlacements } from '../../../types';
+import { Placements } from '../../../constants';
+
```
Comment (crishpeen) [inline packages/web-react/src/components/Dropdown/stories/Dropdown.stories.tsx:6]: why the empty row?

File: packages/web-react/src/components/Tooltip/stories/Tooltip.stories.tsx [hunk 7]
```diff
@@ -2,6 +2,8 @@ import React from 'react';
 import { Markdown } from '@storybook/blocks';
 import type { Meta, StoryObj } from '@storybook/react';
 
+import { Placements } from '../../../constants';
+
```
Comment (crishpeen) [inline packages/web-react/src/components/Tooltip/stories/Tooltip.stories.tsx:6]: why the empty row?

File: packages/web-react/src/components/Tooltip/stories/UncontrolledTooltip.stories.tsx [hunk 8]
```diff
@@ -1,6 +1,8 @@
 import React from 'react';
 import type { Meta, StoryObj } from '@storybook/react';
 
+import { Placements } from '../../../constants';
+
```
Comment (crishpeen) [inline packages/web-react/src/components/Tooltip/stories/UncontrolledTooltip.stories.tsx:5]: why the empty row?

**Review by crishpeen — APPROVED** [review]

File: packages/web-react/src/components/Dropdown/stories/Dropdown.stories.tsx (hunk 6 above)
Comment (literat) [reply packages/web-react/src/components/Dropdown/stories/Dropdown.stories.tsx]: Fixed

File: packages/web-react/src/components/Tooltip/stories/Tooltip.stories.tsx (hunk 7 above)
Comment (literat) [reply packages/web-react/src/components/Tooltip/stories/Tooltip.stories.tsx]: Fixed.

File: packages/web-react/src/components/Tooltip/stories/UncontrolledTooltip.stories.tsx (hunk 8 above)
Comment (literat) [reply packages/web-react/src/components/Tooltip/stories/UncontrolledTooltip.stories.tsx]: Fixed.

File: packages/web-react/src/components/Dropdown/__tests__/useDropdownStyleProps.test.ts (hunk 5 above)
Comment (literat) [reply packages/web-react/src/components/Dropdown/__tests__/useDropdownStyleProps.test.ts]: Fixed.

File: packages/web-react/src/utils/string.ts (hunk 4 above)
Comment (literat) [reply packages/web-react/src/utils/string.ts]: Fixed.

File: packages/web-react/src/components/Tooltip/demo/TooltipPlacements.tsx (hunk 3 above)
Comment (literat) [reply packages/web-react/src/components/Tooltip/demo/TooltipPlacements.tsx]: Fixed.

File: packages/web-react/src/components/Dropdown/useDropdownStyleProps.ts (hunk 2 above)
Comment (literat) [reply packages/web-react/src/components/Dropdown/useDropdownStyleProps.ts]: Fixed.

File: packages/web-react/src/components/Dropdown/Dropdown.tsx (hunk 1 above)
Comment (literat) [reply packages/web-react/src/components/Dropdown/Dropdown.tsx]: Fixed.

**Review by literat — APPROVED** [review]

**Review by crishpeen — APPROVED** [review]

**Pull request merged**